src/backend/distributed/commands/vacuum.c

```diff
@@ -132,107 +139,69 @@ PostprocessVacuumStmt(VacuumStmt *vacuumStmt, const char *vacuumCommand)
 static List *
 VacuumCitusRelationIdList(VacuumStmt *vacuumStmt, CitusVacuumParams vacuumParams)
 {
-	LOCKMODE lockMode = (vacuumParams.options & VACOPT_FULL) ? AccessExclusiveLock :
-						ShareUpdateExclusiveLock;
-
-	List *vacuumRelationList = ExtractVacuumTargetRels(vacuumStmt);
-
 	List *relationIdList = NIL;
 
-	RangeVar *vacuumRelation = NULL;
-	foreach_ptr(vacuumRelation, vacuumRelationList)
+	if (list_length(vacuumStmt->rels) == 0)
+	{
+		/* VACUUM commands without specifiying any table apply to all the tables */
+		relationIdList = CitusTableTypeIdList(ANY_CITUS_TABLE_TYPE);
```

Comment:
Wouldn't it be preferable to send a plain `VACUUM;` to each node if the user did a plain `VACUUM;`?

Reply:
Yes, it is more logical to send a plain VACUUM, I had error previously because worker nodes again try to propagate the plain vacuum after they get it from the coordinator. But now I fixed the problem sending the trio of "disable_ddl_prop; vacuum; enable_ddl_prop;" to workers.